### ISSUE TITLE
Create Block: Add automated tests to scaffold

### ISSUE THREAD

**Issue by aaronjorbin** [WordPress/gutenberg#21360]
**Is your feature request related to a problem? Please describe.**
In order to encourage users to write automated tests, the infrastructure for them should be included in the scaffold from create-block

**Describe the solution you'd like**
Add an option for the esnext template that sets up automated tests that defaults to `Y`.

**Describe alternatives you've considered**
The main alternative is documentation on how to add automated tests. While I think that's a good first step, scaffolding will ultimetely save time. Create-block is installed 70 times a week right now, if we save 5 minutes for each of those people, almost 6 hours a week will be saved. Across an entire year, that is over 12 days of time. 

I also considered making it mandatory, but allowing it to be opted out makes sense to me since there is the possibility that someone may have no interest in automated tests or intends to structure them in a different way.

**Comment by gziolo** [comment]
i totally agree with the proposal. Are you talking about unit tests or/end e2e tests? The latter are a bit harder to set up but it should be all quite straightforward once `@wordpress/env` package matures.

I like the idea of making it possible to opt out. The easiest approach would be to list all commands ( including `lint:js` or `format:js` and let users uncheck some of options.

How should the example test look like?

**Comment by aaronjorbin** [comment]
I was initially thinking unit tests. While not as valuable as e2e, I think they are more straightforward and a little less advanced for engineers to get started writing. The create react app base test does an example that shows `getByText` and a `toBeInDocument` expectation which I think does a good job of showing the type of thing that could be tested. So something like that? 

Though just to get this going, I was thinking of coding it similiar to [the paragraph test](https://github.com/WordPress/gutenberg/blob/master/packages/block-library/src/paragraph/test/edit.native.js) which just tests if it loads. 

The list all commands and uncheck some sounds interesting. What would you think of that being hidden behind something like an advanced config option? Like you need to pass in `--advanced` or something similiar to get it. Ideally the scaffold outputs what the gutenberg team views as the best possible base.

**Comment by gziolo** [comment]
> I was initially thinking unit tests. While not as valuable as e2e, I think they are more straightforward and a little less advanced for engineers to get started writing. The create react app base test does an example that shows `getByText` and a `toBeInDocument` expectation which I think does a good job of showing the type of thing that could be tested. So something like that?

It makes sense to add unit tests first and defer e2e tests setup. The latter probably shouldn't be enabled by default as well as it's an advanced usage that requires `@wordpress/env` integration.

> Though just to get this going, I was thinking of coding it similiar to [the paragraph test](https://github.com/WordPress/gutenberg/blob/master/packages/block-library/src/paragraph/test/edit.native.js) which just tests if it loads.

This example looks complex and it won't scale as it is tested outside of the block editor that provides tons of logic for `edit` implementation. It should be more straightforward to provide a test for `save` method that can be run in isolation using [`getSaveContent`](https://github.com/WordPress/gutenberg/tree/master/packages/blocks#getSaveContent). I would have to double-check that though.

> The list all commands and uncheck some sounds interesting. What would you think of that being hidden behind something like an advanced config option? Like you need to pass in `--advanced` or something similiar to get it. Ideally the scaffold outputs what the gutenberg team views as the best possible base.

I like the idea of `--advanced` option to have all prompts triggered but keeping the list reasonable short otherwise ❤️
